macros/cargo

```diff
@@ -44,5 +47,6 @@ rustflags = %{__global_rustflags_static_toml}\
 EOF\
 )
 
-%cargo_build %{__cargo_cross_env} %{__cargo} build %{__cargo_cross_opts} --release %{?cargo_args}
+%cargo_build %{?_cross_aws_lc_rs:%{set_cross_aws_lc_rs_env}} %{__cargo_cross_env}  %{__cargo} build %{__cargo_cross_opts} --release %{?cargo_args}
+%cargo_build_fips %{?_cross_aws_lc_rs:%{set_cross_aws_lc_rs_env}} %{__cargo_cross_env_fips}  %{__cargo} build %{__cargo_cross_opts} --release %{?cargo_args}
```

Comment:
Can we set something like `--features fips` in the `cargo` arguments to force the use of a FIPS feature, or error out if the target package doesn't understand it?

To me, `%cargo_build_fips` implies a stronger commitment to generating a binary that uses FIPS crypto, like `%cargo_build_static` does for static binaries. Just writing the artifacts to a different directory feels very weak, and it may confuse developers that expect it to do more than that.

Reply:
We could, but AFAIU the `--features` flag can only be set once in `cargo` so we won't be able to override them if we ever have more `cargo` features. We could have a `_cargo_features` macro and for `cargo_build_fips` we could do `--features fips %{_cargo_features}`.

Reply:
Resolved this in offline conversation, we will go with `--features fips` enforced in the macro.